api/tests/mocha/services/generate-xform.spec.js

```diff
@@ -45,24 +63,14 @@ describe('generate-xform service', () => {
       });
     };
 
-    const runTest = (dirname, err) => {
-      const spawned = {
-        stdout: { on: sinon.stub() },
-        stderr: { on: sinon.stub() },
-        stdin: {
-          setEncoding: sinon.stub(),
-          write: sinon.stub(),
-          end: sinon.stub()
-        },
-        on: sinon.stub()
-      };
+    const runTest = (dirname, spawned, err, successClose = true) => {
       sinon.stub(childProcess, 'spawn').returns(spawned);
       return setup(dirname).then(files => {
         const generate = service.generate(files.xform);
         if (err) {
           spawned.stderr.on.args[0][1](err);
           spawned.on.args[0][1](100);
-        } else {
+        } else if (successClose) {
```

Comment:
1. I'm not understanding fully when reading `successClose`, what about `processStout` or `processOutputFlow`
2. so is this running even when we're rejecting [here](https://github.com/medic/cht-core/pull/7301/files#diff-20d679ce4104b549b0cc13a86e91fc38e7fe2d3f3ef3af18236b15cf4dd4b3abR134)?

Reply:
Yes because this is a "spawned" process mock, and there are different places when it can fails: when it needs to read from the input stream, when it needs to write to the output stream, when it needs to close the outstream... so while the first `err` argument makes the mock to fail emulating a writting in the stderr stream as when the real process write to that same stream when the XLS form is invalid, the `successClose` does not compute an error in the process but prevent the mock to return a 0 exit code (success), but the failure is provoked by any of the other mocks in the spawned object.